telemetry/telemetry.go

```diff
@@ -211,6 +213,57 @@ func isFlagPassed(fs *flag.FlagSet, name string) bool {
 	return found
 }
 
+func iNotifyCertMonitoring(watcher *fsnotify.Watcher, telemetryCfg *TelemetryConfig, serverControlSignal chan<- int, testReadySignal chan<- int) {
+	defer watcher.Close()
+
+	done := make(chan bool)
+	go func() {
+		if testReadySignal != nil { // for testing only
+			testReadySignal <- 0
+		}
+		for {
+			select {
+			case event := <-watcher.Events:
+				if event.Name != "" {
+					log.V(1).Infof("Inotify watcher has received event: %v", event)
+					if event.Op & fsnotify.Write == fsnotify.Write {
+						log.V(1).Infof("Cert File has been modified: %s", event.Name)
+						serverControlSignal <- 1 // tells gnmi server to stop and restart with new certs
```

Comment:
>1 [](http://example.com/codeflow?start=29&length=1)

magic numbers. In future if more enum added, it's not easy to understand the code.

Reply:
Will use enums to make code easier to understand.